src/neo/IO/Caching/DataCache.cs

```diff
@@ -145,29 +145,31 @@ public void Delete(TKey key)
         protected abstract void DeleteInternal(TKey key);
 
         /// <summary>
-        /// Find the entries that start with the `key_prefix`
+        /// Seek to the entry with specific key
         /// </summary>
-        /// <param name="key_prefix">Must maintain the deserialized format of TKey</param>
-        /// <returns>Entries found with the desired prefix</returns>
-        public IEnumerable<(TKey Key, TValue Value)> Find(byte[] key_prefix = null)
+        /// <param name="keyOrPrefix">The key to be sought</param>
+        /// <param name="direction">The direction of seek</param>
+        /// <returns>An enumerator containing all the entries after seeking.</returns>
+        public IEnumerable<(TKey Key, TValue Value)> Seek(byte[] keyOrPrefix = null, SeekDirection direction = SeekDirection.Forward)
         {
             IEnumerable<(byte[], TKey, TValue)> cached;
             HashSet<TKey> cachedKeySet;
+            ByteArrayComparer comparer = direction == SeekDirection.Forward ? ByteArrayComparer.Default : ByteArrayComparer.Reverse;
             lock (dictionary)
             {
                 cached = dictionary
-                    .Where(p => p.Value.State != TrackState.Deleted && (key_prefix == null || p.Key.ToArray().AsSpan().StartsWith(key_prefix)))
+                    .Where(p => p.Value.State != TrackState.Deleted && (keyOrPrefix == null || comparer.Compare(p.Key.ToArray(), keyOrPrefix) >= 0))
```

Comment:
If keyOrPrefix is null and direction is backward, you will find nothing. But it should find everything.